asmcomp/amd64/emit.mlp

```diff
@@ -786,8 +786,11 @@ let emit_instr fallthrough i =
   | Lop(Ispecific(Ibswap _)) ->
       assert false
   | Lop(Ispecific Isqrtf) ->
+      if arg i 0 <> res i 0 then
```

Comment:
I know nothing about the topic, but from your description, it's not obvious to me why clearing the destination register couldn't be beneficial when it is the same as the argument.

Reply:
Because that would result in zero, regardless of the input value! (I had this bug in the first version of this patch)

Reply:
And there's no "dependency breaking idiom" for zeroing the upper half only ?

Reply:
There doesn't need to be. `sqrtsd %xmm0, %xmm0` does depend on the previous value of `%xmm0`. It is correct for the processor to wait for it to be computed.